### PR TITLE
Get default path to openvpn.exe and openvpn-gui.exe from registry

### PR DESCRIPTION
**Pull request by mattock**
URL: https://github.com/OpenVPN/openvpn-windows-test/pull/3
Signed-off-by: Samuli Seppänen <samuli@openvpn.net>

### CODE REVIEW

File: test-openvpn.ps1 [hunk 1]
```diff
@@ -35,12 +35,16 @@ Function Verify-Path ([string]$mypath) {
...
 if (! $Openvpn) {
-    $Openvpn = "C:\Program Files\OpenVPN\bin\openvpn.exe"
+    $Openvpn = $exe_path
 }
 
 if (! $Gui) {
-    $Gui = "C:\Program Files\OpenVPN\bin\openvpn-gui.exe"
+    # Assume openvpn-gui.exe is in the same directory as openvpn.exe
+    $exe_dir = $exe_path.TrimEnd('openvpn.exe')
+    $Gui = "${exe_dir}openvpn-gui.exe"
```
Comment (chipitsine) [inline test-openvpn.ps1]: I'd use JOin-Path for that.

like 

$Gui = (Join-Path $exe_dir 'openvpn-gui.exe')

it is very rare case, Join-Path takes care of path separator, which is different on Mono and Windows.
I wouldn't beleive we'll be running openvpn on Linux/Mono sometimes :)

Join-Path is considered a safe way of joining paths together

Comment (mattock) [inline test-openvpn.ps1]: Agreed. Force-pushed an update.

File: test-openvpn.ps1 [hunk 2]
```diff
@@ -35,12 +35,16 @@ Function Verify-Path ([string]$mypath) {
...
+$exe_path = (Get-ItemProperty 'HKLM:\SOFTWARE\OpenVPN' -Name 'exe_path').exe_path
+
 if (! $Openvpn) {
-    $Openvpn = "C:\Program Files\OpenVPN\bin\openvpn.exe"
+    $Openvpn = $exe_path
 }
 
 if (! $Gui) {
-    $Gui = "C:\Program Files\OpenVPN\bin\openvpn-gui.exe"
+    # Assume openvpn-gui.exe is in the same directory as openvpn.exe
```
Comment (chipitsine) [inline test-openvpn.ps1]: there's separate registry value for openvpn-gui.exe, shouldn't we take it instead ?

File: test-openvpn.ps1 [hunk 3]
```diff
@@ -35,12 +35,16 @@ Function Verify-Path ([string]$mypath) {
     }
 }
 
+$exe_path = (Get-ItemProperty 'HKLM:\SOFTWARE\OpenVPN' -Name 'exe_path').exe_path
+
 if (! $Openvpn) {
```
Comment (mattock) [inline test-openvpn.ps1]: I agree that that way of handling default values for parameters is cleaner. Most of the time we can safely assume the default values can be obtained (e.g. exe_path exists in registry). I would refactor parameter handling in another PR, though.

I think we should do a minimal environment check quite early on. For example, check if the OpenVPN registry key exists, and if not, fail immediately.

File: test-openvpn.ps1 (hunk 2 above)
Comment (mattock) [inline test-openvpn.ps1]: Which registry key is that? I only see pointers to openvpn.exe.

Comment (chipitsine) [inline test-openvpn.ps1]: hmm, indeed, you are right

**Comment by mattock** [comment]
Approved? :smile:

**Pull request merged**